torch/_dynamo/output_graph.py

```diff
@@ -589,6 +594,25 @@ def register_attr_or_module(
         source = options["source"]
         assert not isinstance(source, ParamBufferSource)
 
+        if is_dynamic_nn_module(target) and getattr(
+            target, "_is_fsdp_managed_module", False
+        ):
+            name = "_".join(map(str, names))
+            base = name
+            for i in itertools.count():
+                if name not in self.nn_modules:
+                    self.nn_modules[name] = target
+                    break
+                name = f"{base}_{i}"
+            options["guards"].add(source.make_guard(GuardBuilder.TYPE_MATCH))
+            options["guards"].add(source.make_guard(GuardBuilder.ID_MATCH))
+            vt = variables.nn_module.FSDPManagedNNModuleVariable(
+                target,
+                name,
+                **options,
+            )
+            return self.side_effects.track_object_existing(source, target, vt)
```

Comment:
is the pattern that top-level graph-input things come here to get registered/wrapped but other things go through builder?  I forgot the details, but its bugging me that we didn't have this case at all before and things seemed to mostly work.

Reply:
Well, before, we would graph break on access here! It was never broken, just not covered.